src/main/java/strategy/handlers/SupplyHandler.java

```diff
@@ -0,0 +1,12 @@
+package strategy.handlers;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class SupplyHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.fruitStorage.merge(transaction.getFruit(), transaction.getQuantity(), Integer::sum);

```

Comment:
Remember to validate the `transaction` object to ensure it's not null before using it to prevent `NullPointerException`. Also, ensure that `transaction.getQuantity()` returns a non-negative value as supplying a negative quantity doesn't make sense in this context.